generator/generator.go

```diff
@@ -72,6 +72,7 @@ var (
 	ignoreShadowSchemaPaths              = flag.Bool("ignore_shadow_schema_paths", false, "If set to true when compress_paths=true, the shadowed schema path will be ignored while unmarshalling instead of causing an error. A shadow schema path is a config or state path which is selected over the other during schema compression when both config and state versions of the node exist. NOTE: Invalid shadow paths currently do not cause an error during JSON unmarshalling.")
 	shortenEnumLeafNames                 = flag.Bool("shorten_enum_leaf_names", false, "If also set to true when compress_paths=true, all leaves of type enumeration will by default not be prefixed with the name of its residing module.")
 	useDefiningModuleForTypedefEnumNames = flag.Bool("typedef_enum_with_defmod", false, "If set to true, all typedefs of type enumeration or identity will be prefixed with the name of its module of definition instead of its residing module.")
+	appendEnumSuffixForSimpleUnionEnums  = flag.Bool("enum_suffix_for_simple_union_enums", false, "If set to true when typedef_enum_with_defmod is also true, all inlined enumerations within unions will be suffixed with \"Enum\", instead of adding the suffix only for inlined enumerations within typedef unions.")
```

Comment:
At this point, I'm wondering whether we just deprecate all these in v1.0.0 and cut this a bit further down the line (Q2 '21 maybe?). It seems like we can get rid of all three of these flags then.

Reply:
(no action required)

Reply:
Yep that's what I'm thinking too -- this is not only complexity for the users but also a maintenance burden.